### PR TITLE
Default normalization dropdown vsi

### LINKED ISSUE #13601: Create default normalization dropdown for VSI
We want a normalization dropdown for the VSI. There the user can specify a default normalization.

This is merely an alias to the normalization selection field in each filter properties. However, since we aim to eventually hide most properties and the pipeline browser, it would be nice to sync this property to a more prominent location.

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Fixes #13601 

**Context**

The functionality allows the user to select a default inital normalization for when a new MDEventWorkspace or MDHistoWorksapce is loaded . The selection is presisted and available in the following session.
# For testing

You will need MDHisto and MDEvent sample workspaces which can be found here: \olympic\Babylon5\Scratch\Anton\VSI\SAMPLE_WORKSPACES

A OSX disk image of the Jenkins build can be found here: \olympic\Babylon5\Scratch\Anton\VSI\13626_VSI_NORM_DROPDOWN

Perform all test once for MDHisto and once for MDEvent workspaces.

**Test that the initial normalization is being applied**
1. Load the workspace into the VSI
2. Select from the appropriate dropdown menu the default initial normalization "None"
3. Close the VSI and reload the workspace
   - Confirm that the normalization in the properties browser matches the one which was selected in the dropdown menu
4. Repeat the last two steps with any setting and confirm that the selection in the properties browser matches the dropdown setting.
5. Make a note of the dropdown setting.
6. Close and restart Mantid, load the data set into the VSI
   - Confirm that the dropdown setting is the same as before Mantid was closed.

### CODE REVIEW

**Comment by quantumsteve** [comment]
@picatess I haven't tried your branch, but from the description this pull request appears to work against issue #12832. In particular, 

* The choice of normalization depends on more than the type of workspace.
* We don't want the user choosing a potentially incorrect normalization option.

**Comment by picatess** [comment]
@quantumsteve  This is not the PR for #12832. This is a just a GUI enhancement for the VSI. The user can select his/her perferred normalization selection. This is something @OwenArnold  requested and should be connected with the solution for #12832. I will mark it as "in pogress", then it can be adjusted later on (It is currently marked as 3.6)

**Comment by peterfpeterson** [comment]
Let's wait until Owen gets back and discus if it is still appropriate if the other solution can come soon enough.

**Comment by OwenArnold** [comment]
@picatess  See general comments against issue.

File: Code/Mantid/Framework/MDAlgorithms/src/SliceMD.cpp [hunk 1]
```diff
@@ -285,7 +285,10 @@ void SliceMD::slice(typename MDEventWorkspace<MDE, nd>::sptr ws) {
   // output event workspace
   IMDEventWorkspace_sptr outEvent =
       boost::dynamic_pointer_cast<IMDEventWorkspace>(outWS);
-  outEvent->setDisplayNormalization(ws->displayNormalization());
+  // Since the output of the SliceMD algorithm has regular binning we set the
+  // display normalization
+  // of the MDEvent output workspace to displayNormalizationHisto.
+  outEvent->setDisplayNormalization(ws->displayNormalizationHisto());
```
Comment (OwenArnold) [inline Code/Mantid/Framework/MDAlgorithms/src/SliceMD.cpp]: This is good.

File: Code/Mantid/MantidQt/API/src/MdConstants.cpp [hunk 2]
```diff
@@ -52,6 +52,11 @@ namespace MantidQt
       m_multiSliceView = "Multi Slice";
       m_threeSliceView = "Three Slice";
       m_splatterPlotView = "Splatter Plot";
+
+      m_autoNormalization = "Auto";
```
Comment (OwenArnold) [inline Code/Mantid/MantidQt/API/src/MdConstants.cpp]: I think this should be called Auto Select (see Owen for more info)

File: Code/Mantid/MantidQt/API/src/MdConstants.cpp [hunk 3]
```diff
@@ -52,6 +52,11 @@ namespace MantidQt
       m_multiSliceView = "Multi Slice";
       m_threeSliceView = "Three Slice";
```
Comment (OwenArnold) [inline Code/Mantid/MantidQt/API/src/MdConstants.cpp]: Any reason not to use static consts?

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/inc/MantidVatesSimpleGuiViewWidgets/MdViewerWidget.ui [hunk 4]
```diff
@@ -108,6 +111,26 @@
        </property>
       </spacer>
      </item>
+     <item>
+      <widget class="QLabel" name="lbl_normalizationHisto">
+       <property name="toolTip">
+        <string extracomment="Select the default initial normalization when MDHisto workspaces are loaded"/>
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/inc/MantidVatesSimpleGuiViewWidgets/MdViewerWidget.ui]: Should all be in bespoke widget, not in MDViewerWidget

**Comment by OwenArnold** [comment]
@picatess Do you want to remerge master into this branch. I think it's probably stale, I'll verify it asap after.

**Comment by picatess** [comment]
@OwenArnold  I have been working on your suggestions ( which has everythgin updated). I am not sure if it makes much sense to go ahead with this feature before we have decided if we will have single instances or not, as this will change the way we have to implement the NormalizationWidget(or at least the underlying pipeworks). It makes probably sense to delete the PR for now

**Comment by OwenArnold** [comment]
@picatess Fine. You may want to ensure you keep a local copy of your branch around.

**Pull request closed without merging**